src/Task/Git/BranchName.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace GrumPHP\Task\Git;
+
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Util\Regex;
+use GrumPHP\Exception\RuntimeException;
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Task\TaskInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Gitonomy\Git\Repository;
+
+/**
+ * Git BranchName Task
+ */
+class BranchName implements TaskInterface
+{
+
+    /**
+     * @var GrumPHP
+     */
+    protected $grumPHP;
+
+    /**
+     * @var Repository
+     */
+    protected $repository;
+
+    /**
+     * @param GrumPHP $grumPHP
+     */
+    public function __construct(GrumPHP $grumPHP, Repository $repository)
+    {
+        $this->grumPHP = $grumPHP;
+        $this->repository = $repository;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'git_branch_name';
+    }
+
+    /**
+     * @return array
+     */
+    public function getConfiguration()
+    {
+        $configured = $this->grumPHP->getTaskConfiguration($this->getName());
+
+        return $this->getConfigurableOptions()->resolve($configured);
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+          'matchers' => [],
+          'additional_modifiers' => ''
+        ]);
+
+        $resolver->addAllowedTypes('matchers', ['array']);
+        $resolver->addAllowedTypes('additional_modifiers', ['string']);
+
+        return $resolver;
+    }
+
+    /**
+     * @param ContextInterface $context
+     *
+     * @return bool
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return $context instanceof RunContext || $context instanceof GitPreCommitContext;
+    }
+
+    /**
+     * @param array $config
+     * @param string $name
+     * @param string $rule
+     * @param string $ruleName
+     *
+     * @throws RuntimeException
+     */
+    private function runMatcher(array $config, $name, $rule, $ruleName)
+    {
+        $regex = new Regex($rule);
+
+        $additionalModifiersArray = array_filter(str_split($config['additional_modifiers']));
+        array_map([$regex, 'addPatternModifier'], $additionalModifiersArray);
+
+        if (!preg_match((string) $regex, $name)) {
+            throw new RuntimeException("Rule not matched: \"$ruleName\" $rule");
+        }
+    }
+
+    /**
+     * @param ContextInterface|RunContext $context
+     *
+     * @return TaskResult
+     */
+    public function run(ContextInterface $context)
+    {
+        $name = trim($this->repository->run('symbolic-ref', ['HEAD', '--short']));
```

Comment:
When the current revision is a non symbolic reference, the command will throw an exception like: 
> Error code: 128
> Message: fatal: ref HEAD is not a symbolic ref

This can happen in detached head or if you checked out a specific commit.
This will throw a `Gitonomy\Git\Exception\ProcessException`.

How would you like to handle this issue on the task? Should it be configurable to allow non symbolic references or not? We might want to fetch the error message and display a user-friendly error message instead?